views/js/payment-components/one-click.js

```diff
@@ -41,22 +40,31 @@ jQuery(function ($) {
 
     var threeDSProcessUrl;
 
-    $('[data-one-click-payment]').each(function (index, element) {
-        var oneClickPaymentConfiguration = $(element).data();
-        var oneClickPaymentMethod = oneClickPaymentConfiguration.oneClickPayment;
-        if (oneClickPaymentMethod) {
-            renderOneClickComponent(oneClickPaymentMethod);
-            fillBrowserInfo();
+    // use this object to iterate through the stored payment methods
+    var checkoutStoredPaymentMethods = window.adyenCheckout.paymentMethodsResponse.storedPaymentMethods;
+
+    // Iterate through the stored payment methods list we got from the adyen checkout component
+    checkoutStoredPaymentMethods.forEach(function (storedPaymentMethod) {
+
+        /*
+            storedPaymentMethod.id = $storedPaymentApiId in the oneclick.tpl
+            if the container doesn't exits don't try to render the component
+        */
+        var storedPaymentMethodContainer = $("#cardContainer-" + storedPaymentMethod.id);
+        if (!storedPaymentMethodContainer.length) {
+            console.log("Something went wrong with the payment method");
+            return;
         }
 
-        threeDSProcessUrl = oneClickPaymentConfiguration.threeDsProcessUrl;
+        var storedPaymentMethodConfiguration = $('[data-stored-payment-api-id="' + storedPaymentMethod.id + '"]').data();
```

Comment:
```suggestion
        var storedPaymentMethodConfiguration = storedPaymentMethodContainer.find('[data-stored-payment-api-id="' + storedPaymentMethod.id + '"]').data();
```
I don't know if this is possible, but would be better to restrict the search to the container

Reply:
unfortunately the div that we want to find is outside of the storedPaymentMethodContainer so we can relay on the storedPaymentMethod.id